### DISCUSSION TITLE
Rust, are You that serious??? :)

### DISCUSSION THREAD

**Discussion by SarK0Y** [community/community#118501] [Programming Help]
### Select Topic Area

Bug

### Body

https://github.com/SarK0Y/TAM_RUSTy/commit/474728c45f6685d950d14260cebe6f0dff6cb276

fish: Job 1, 'cargo run -- -find-files -pat...' terminated by signal SIGSEGV (Address boundary error)
come on, Rust, are You serious???😜😇🤣🤣

**Comment by jfullstackdev** [comment]
Copilot maybe of help, 

---

The error message `terminated by signal SIGSEGV (Address boundary error)` typically indicates a segmentation fault. This happens when a program tries to access a memory location that it's not allowed to access, such as a null pointer, an area of memory reserved by the operating system, or an area of memory that the program has already freed.

Here are some steps you can take to debug this issue:

1. **Check for null pointers**: Make sure you're not dereferencing null pointers anywhere in your code.

2. **Check array bounds**: If you're accessing arrays, make sure you're not going beyond their bounds.

3. **Use a debugger**: Tools like `gdb` can help you identify the exact line of code that's causing the segmentation fault.

4. **Use Rust's built-in tools**: Rust has some built-in tools that can help with debugging, such as the `debug!` macro and the `#[track_caller]` attribute.

5. **Check your dependencies**: If you're using third-party libraries, make sure they're up to date and don't have any known issues that could be causing this problem.

Without more specific information about your code, it's hard to provide more detailed advice. If you're still having trouble, please provide more details about your code and the context in which this error is occurring.

**Reply by SarK0Y** [reply]
the very point is, Rust should have caught this error.. how to debug it isn't a problem, but anyway thanks for reply :)

**Comment by LiteBrite82** [comment]
Thanks for posting in the GitHub Community, @SarK0Y!

We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Discussions category is solely related to conversations around the [GitHub product Discussions](https://docs.github.com/en/discussions/collaborating-with-your-community-using-discussions/about-discussions). This question should be in the Programming Help category. I've gone ahead and moved it for you. Good luck!
